src/components/Dropdown/Dropdown.tsx

```diff
@@ -162,20 +181,49 @@ function useFocusContextInstance(
   lastTypedOrSubmittedValue: string,
   setValue: (newValue: string) => void,
   screenReaderKey: number,
-  setScreenReaderKey: (newKey: number) => void
+  setScreenReaderKey: (newKey: number) => void,
+  alwaysSelectOption: boolean,
+  selectedBeforeClose: boolean
 ): FocusContextType {
   const [focusedIndex, setFocusedIndex] = useState(-1);
   const [focusedValue, setFocusedValue] = useState<string | null>(null);
   const [focusedItemData, setFocusedItemData] = useState<Record<string, unknown> | undefined>(undefined);
+  useEffect(() => {
+    if (alwaysSelectOption) {
+      if (items.length > 0) {
+        const index = focusedIndex === -1 || focusedIndex > items.length ? 0 : focusedIndex;
+        setFocusedIndex(index);
+        setFocusedValue(items[index].value);
+        setFocusedItemData(items[index].itemData);
+      } else {
+        setFocusedIndex(-1);
+        setFocusedValue(null);
+        setFocusedItemData(undefined);
+      }
+      if (!selectedBeforeClose) {
+        setFocusedIndex(-1);
+        setFocusedValue(null);
+        setFocusedItemData(undefined);
+      }
+    }
+  }, [alwaysSelectOption, focusedIndex, items, selectedBeforeClose]);
 
   function updateFocusedItem(updatedFocusedIndex: number, value?: string) {
     const numItems = items.length;
     let updatedValue;
+    let updateFocusedValue = true;
     if (updatedFocusedIndex === -1 || updatedFocusedIndex >= numItems || numItems === 0) {
       updatedValue = value ?? lastTypedOrSubmittedValue;
-      setFocusedIndex(-1);
-      setFocusedItemData(undefined);
-      setScreenReaderKey(screenReaderKey + 1);
+      if (alwaysSelectOption && numItems !== 0 && selectedBeforeClose) {
+        setFocusedIndex(0);
+        setFocusedItemData(items[0].itemData);
+        setScreenReaderKey(screenReaderKey + 1);
+        updateFocusedValue = false;
+      } else {
+        setFocusedIndex(-1);
+        setFocusedItemData(undefined);
+        setScreenReaderKey(screenReaderKey + 1);
```

Comment:
is there a reason why we only sometimes call `setScreenReaderKey` in various cases in updateFocusedItem? as well as not calling it in the useEffect above?